### PR TITLE
Handle numeric columns

### PR DESCRIPTION
**Pull request by nick863**
# Description

Handle the numeric columns in the evaluator output. See work item 3216567

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 1]
```diff
@@ -395,14 +395,14 @@ def evaluate(
 
         # drop input columns
         evaluator_result_df = evaluator_result_df.drop(
-            columns=[col for col in evaluator_result_df.columns if col.startswith(Prefixes._INPUTS)]
+            columns=[col for col in evaluator_result_df.columns if str(col).startswith(Prefixes._INPUTS)]
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:398]: this is a good conner case can be covered by unit test. if takes time, pls create an ado for now.

**Review by ninghu — APPROVED** [review]

**Review by singankit — DISMISSED** [review]

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py (hunk 1 above)
Comment (nick863) [reply src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:402]: Fixed

File: src/promptflow-evals/tests/evals/e2etests/test_evaluate.py [hunk 2]
```diff
@@ -132,15 +145,26 @@ def test_evaluate_with_content_safety_evaluator(self, project_scope, data_file):
         assert 0 <= metrics.get("content_safety.self_harm_defect_rate") <= 1
         assert 0 <= metrics.get("content_safety.hate_unfairness_defect_rate") <= 1
 
-    @pytest.mark.parametrize('use_thread_pool', [True, False])
-    def test_evaluate_python_function(self, data_file, use_thread_pool):
+    @pytest.mark.parametrize('use_thread_pool,function,column', [
+        (True, answer_evaluator, 'length'),
+        (False, answer_evaluator, 'length'),
+        (True, answer_evaluator_json, 'length'),
+        # PFClient does not deserialize the JSON.
```
Comment (ninghu) [inline src/promptflow-evals/tests/evals/e2etests/test_evaluate.py:152]: I think we should keep the behavior same as PFClient.

File: src/promptflow-evals/promptflow/evals/evaluate/_code_client/code_client.py [hunk 3]
```diff
@@ -63,7 +63,16 @@ def _calculate_metric(self, evaluator, input_df, column_mapping, evaluator_name)
...
-                row_metric_results.append(row_metric_future.result())
+                result = row_metric_future.result()
+                if isinstance(result, str):
+                    try:
+                        result = json.loads(result)
+                    except BaseException:
+                        # Handle in the next section.
+                        pass
+                if not isinstance(result, dict):
+                    result = {'output': result}
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_code_client/code_client.py:74]: Does "output" also PFClient returns?

File: src/promptflow-evals/promptflow/evals/evaluate/_code_client/code_client.py [hunk 4]
```diff
@@ -63,7 +63,16 @@ def _calculate_metric(self, evaluator, input_df, column_mapping, evaluator_name)
 
         for row_number, row_metric_future in enumerate(row_metric_futures):
             try:
-                row_metric_results.append(row_metric_future.result())
+                result = row_metric_future.result()
+                if isinstance(result, str):
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_code_client/code_client.py:67]: What if user just wanted to return a json string?

File: src/promptflow-evals/tests/evals/e2etests/test_evaluate.py (hunk 2 above)
Comment (ninghu) [reply src/promptflow-evals/tests/evals/e2etests/test_evaluate.py:152]: Thoughts?

**Review by ninghu — APPROVED** [review]

**Review by singankit — APPROVED** [review]

**Pull request merged**